pandera/dtypes.py

```diff
@@ -358,6 +376,41 @@ class Complex64(Complex128):
     bit_width = 64
 
 
+###############################################################################
+# decimal
+###############################################################################
+
+
+@immutable(init=True)
+class Decimal(_Number):
+    """Semantic representation of a decimal data type."""
+
+    exact: bool = dataclasses.field(init=False, default=True)
+    continuous: bool = dataclasses.field(init=False, default=True)
+
+    precision: Optional[int] = None
+    scale: Optional[int] = None
+
+    def __init__(
+        self, precision: Optional[int] = None, scale: Optional[int] = None
+    ):
+        super().__init__()
+        if precision is not None:
+            if precision <= 0:
+                raise ValueError(
+                    f"Decimal precision {precision} must be positive."
+                )
+            if scale is not None and scale > precision:
+                raise ValueError(
+                    f"Decimal scale {scale} must be between 0 and {precision}."
```

Comment:
can we add test cases for the ValueErrors raised by these two execution paths?

Reply:
yes !